### PR TITLE
Root fix

### PR DESCRIPTION
**Pull request by CJ-Wright**
In which we find and fix the root of the problem for change root

### CODE REVIEW

File: portable_fs/template/fs.py [hunk 1]
```diff
@@ -236,10 +236,12 @@ def get_spec_handler(self, resource):
...
         rpath = resource['resource_path']
         root = resource.get('root', '')
-
+        root = self.root_map.get(root, root)
         if root:
-            root = self.root_map.get(root, root)
+            if os.path.isabs(rpath):
+                rpath = rpath[1:]
             rpath = os.path.join(root, rpath)
+        print(rpath)
```
Comment (tacaswell) [inline portable_fs/template/fs.py]: debugging print statement.

File: portable_fs/template/fs.py [hunk 2]
```diff
@@ -236,10 +236,12 @@ def get_spec_handler(self, resource):
         kwargs = resource['resource_kwargs']
         rpath = resource['resource_path']
         root = resource.get('root', '')
-
+        root = self.root_map.get(root, root)
         if root:
-            root = self.root_map.get(root, root)
+            if os.path.isabs(rpath):
+                rpath = rpath[1:]
```
Comment (tacaswell) [inline portable_fs/template/fs.py]: Not sure this is portable across OSes?

Comment (CJ-Wright) [reply portable_fs/template/fs.py]: That is potentially true. I will look into os.path for a way to remove the drive/root

Comment (CJ-Wright) [reply portable_fs/template/fs.py]: I can get rid of this with pathlib

**Review by tacaswell — APPROVED** [review]

**Comment by CJ-Wright** [comment]
@tacaswell in an slightly related note to our conversation today, I think this is ready for merge.

File: portable_fs/template/fs.py [hunk 3]
```diff
@@ -11,6 +11,7 @@
 from . import core
 from collections import defaultdict
 import warnings
+import pathlib
```
Comment (danielballan) [inline portable_fs/template/fs.py]: This is Python 3 only. Need to add pathlib backport package as a Python 2 dependency.

Comment (CJ-Wright) [reply portable_fs/template/fs.py]: Ah, but you don't build against python 2.7

Comment (CJ-Wright) [reply portable_fs/template/fs.py]: I assume that if you are not running [CI against py2k it is not supported](https://github.com/NSLS-II/portable-fs/blob/master/.travis.yml#L8)

Comment (danielballan) [reply portable_fs/template/fs.py]: That's a reasonable assumption. As you know these are toy projects, which is why CI testing is not rigorously set up and some py3-only code slipped through. We're moving all these into the broker repo soon-ish, so the content of this PR will need to be refactored to be py2 compatible either now or later. Better to do it now so it's not one more test failure to fix later.

**Comment by CJ-Wright** [comment]
This is failing because of python 3->2 `super()` problems. Can we merge this? I don't want to do a full 3->2 backport in this PR (scope creep?)

**Comment by tacaswell** [comment]
Can you remove the 2.7 entry from the test matrix or do the patch up?

**Comment by CJ-Wright** [comment]
Thank you!

**Pull request merged**